src/css-selector-generator.coffee

```diff
@@ -74,27 +131,33 @@ class CssSelectorGenerator
       # trim whitespace
       class_string = class_string.replace /^\s|\s$/g, ''
       if class_string isnt ''
-        result = for item in class_string.split /\s+/
-          ".#{@sanitizeItem item}"
+        for item in class_string.split /\s+/
+          if @notInList item, @options.class_blacklist
+            result.push ".#{@sanitizeItem item}"
     result
 
   getAttributeSelectors: (element) ->
     result = []
-    blacklist = ['id', 'class']
-    for attribute in element.attributes
-      unless attribute.nodeName in blacklist
-        result.push "[#{attribute.nodeName}=#{attribute.nodeValue}]"
+    whitelist = @options.attribute_whitelist
+    for attr in whitelist
+      if element.hasAttribute attr
+        result.push "[#{attr}=#{@sanitizeAttribute element.getAttribute(attr)}]"
+    blacklist = @options.attribute_blacklist.concat(['id', 'class'])
```

Comment:
If we get a whitelist, do we want to add other classes to the result?

```suggestion
    whitelist = @options.attribute_whitelist
    if whitelist.length > 0
      for attr in whitelist
        if element.hasAttribute attr
          result.push "[#{attr}=#{@sanitizeAttribute element.getAttribute(attr)}]"
      return result
```

Reply:
I think so. May be the naming is not proper - please see my explanation above.